extensions/libarchive/BinFiles.cpp

```diff
@@ -223,11 +223,14 @@ bool BinFiles::resurrectFlowFiles(core::ProcessSession &session) {
   return had_failure;
 }
 
-void BinFiles::assumeOwnershipOfNextBatch(core::ProcessSession &session) {
+bool BinFiles::assumeOwnershipOfNextBatch(core::ProcessSession &session) {
   for (size_t i = 0; i < batchSize_; ++i) {
     auto flow = session.get();
 
     if (flow == nullptr) {
+      if (i == 0) {
+        return false;
+      }
```

Comment:
besides that, I wanna do a proper fix aswell (we should probably refactor the hasWorkToDo thingy so we can skip the ontrigger part if its not neccesary) I just want to get this PR merged asap as it can cause some heavy resource usage